resources/roca_lib.brs

```diff
@@ -166,6 +177,14 @@ function __case_execute()
         __state: m.__state
         assert: assert(__util_pass, __util_fail, m.__state)
     }
+
+    ' extra case fields
+    if m.suite.__fields <> invalid then
+        for each item in m.suite.__fields.items()
+            withM.addreplace(item.key, item.value)
+        end for
```

Comment:
If `m.suite.__fields` is already an associative array and `withM` is an associative array (and we're comfortable with [allowing users to stomp on required names](If `m.suite.__fields` is already an associative array and `withM` is an associative array, we may as well replace this whole) we may as well replace this whole `for each` loop with `ifAssociativeArray#update` and avoid iteration entirely 😃 

```suggestion
        withM.append(m.suite.__fields)
```

Reply:
I feel like I'm seeing double with this comment 😆

Reply:
> I feel like I'm seeing double with this comment 😆

😬 turns out firefox didn't copy the link to your comment like it said it did 🤦